app/org/sagebionetworks/bridge/dynamodb/DynamoUploadSchemaDao.java

```diff
@@ -82,6 +86,101 @@ public void setStudyIdIndex(DynamoIndexHelper studyIdIndex) {
 
     /** {@inheritDoc} */
     @Override
+    public @Nonnull UploadSchema createUploadSchemaFromSurvey(@Nonnull StudyIdentifier studyIdentifier,
+            @Nonnull Survey survey) {
+        // create upload field definitions from survey questions
+        List<UploadFieldDefinition> fieldDefList = new ArrayList<>();
+        for (SurveyQuestion oneQuestion : survey.getUnmodifiableQuestionList()) {
+            String name = oneQuestion.getIdentifier();
+            UploadFieldType type = getFieldTypeFromConstraints(oneQuestion.getConstraints());
+            UploadFieldDefinition oneFieldDef = new DynamoUploadFieldDefinition.Builder().withName(name)
+                    .withType(type).build();
+            fieldDefList.add(oneFieldDef);
+        }
+
+        // Get the current rev.
+        String studyId = studyIdentifier.getIdentifier();
+        String schemaId = survey.getIdentifier();
+        int oldRev = getCurrentSchemaRevision(studyId, schemaId);
+
+        // Create schema.
+        DynamoUploadSchema schema = new DynamoUploadSchema();
+        schema.setFieldDefinitions(fieldDefList);
+        schema.setName(survey.getName());
+        schema.setRevision(oldRev);
+        schema.setSchemaId(schemaId);
+        schema.setSchemaType(UploadSchemaType.IOS_SURVEY);
+        schema.setStudyId(studyId);
+        UploadSchema createdSchema = createOrUpdateUploadSchema(studyId, schema);
+
+        return createdSchema;
+    }
+
+    /**
+     * Private helper function that converts a survey question constraints object into an upload schema field type.
+     * This is used to help convert surveys into upload schemas.
+     */
+    private static UploadFieldType getFieldTypeFromConstraints(Constraints constraints) {
+        if (constraints == null) {
+            // Could be anything. When in doubt, default to inline_json_blob.
+            return UploadFieldType.INLINE_JSON_BLOB;
+        }
+
+        if (constraints instanceof MultiValueConstraints
+                && ((MultiValueConstraints) constraints).getAllowMultiple()) {
+            if (constraints.getDataType() == DataType.STRING) {
+                // Multiple choice string answers frequently become longer than the 100-char limit. This will have
+                // to be a JSON attachment.
+                return UploadFieldType.ATTACHMENT_JSON_BLOB;
+            } else {
+                // Multiple choice non-string answers (frequently ints) tend to be very short, so this can fit in
+                // an inline_json_blob.
+                return UploadFieldType.INLINE_JSON_BLOB;
+            }
+        } else {
+            switch (constraints.getDataType()) {
```

Comment:
DataType/constraints type should always match, though there's no harm in having fallbacks.